src/include/stir/ProjDataInMemory.h

```diff
@@ -92,20 +95,35 @@ class ProjDataInMemory : public ProjDataFromStream
   virtual void axpby(const float a, const ProjData& x,
                      const float b, const ProjData& y);
 
+  //! start value for iterating through all elements in the array, see full_iterator
+  inline DataType::full_iterator begin_all()
```

Comment:
no need to expose `begin_all` etc just stick to "normal" iterators I believe? Up to you, but would need both

Reply:
So just `begin()` and `end()` with `const` and non-`const` versions? (That's what I've done in the most recent commit anyway.)

Reply:
I think having only `begin()` is fine, but having `begin_all()` as well is nicer :smile:

Reply:
ok, added too.